MediaPlayer/MediaPlayer.js

```diff
@@ -15,37 +19,432 @@ import Times from './Times';
+		 * @public
+		 */
+		source: PropTypes.node.isRequired,
+
+		/**
+		 * The current time in seconds of the media source.
+		 *
+		 * @type {Number}
+		 * @public
+		 */
+		currentTime: PropTypes.number,
+
+		/**
+		 * The current locale as a
+		 * {@link https://tools.ietf.org/html/rfc5646|BCP 47 language tag}.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		locale: PropTypes.string,
+
+		/**
+		 * `true` when the media loops.
+		 *
+		 * @type {Boolean}
+		 * @public
+		 */
+		loop: PropTypes.bool,
+
+		/**
+		 * Media component to use.
+		 *
+		 * The default (`'video'`) renders an `HTMLVideoElement`. Custom media components must have
+		 * a similar API structure, exposing the following APIs:
+		 *
+		 * Methods:
+		 * * `load()` - load media
+		 *
+		 * @type {String|Component}
+		 * @default 'audio'
+		 * @public
+		 */
+		mediaComponent: EnactPropTypes.renderable,
+
+		/**
+		 * Function that generates a reference to the current loaded media
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		mediaRef: PropTypes.func,
+
+		/**
+		 * Called when media is looped
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onLoopChange: PropTypes.func,
+
+		/**
+		 * Called when media is paused
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onPause: PropTypes.func,
+
+		/**
+		 * Called when media is played
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onPlay: PropTypes.func,
+
+		/**
+		 * Called when media is updating.
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onUpdate: PropTypes.func,
+
+		/**
+		 * The media pause state.
+		 *
+		 * @type {Boolean}
+		 * @public
+		 */
+		paused: PropTypes.bool,
+
+		/**
+		 * The total time (duration) in seconds of the loaded media source.
+		 *
+		 * @type {Number}
+		 * @public
+		 */
+		total: PropTypes.number
+	},
+
+	defaultProps: {
+		mediaComponent: 'audio'
+	},
+
 	styles: {
 		css,
 		className: 'mediaPlayer'
 	},
 
-	render: ({...rest}) => {
+	render: ({currentTime, locale, loop, mediaComponent, mediaRef, onLoopChange, onPause, onPlay, onUpdate, paused, source, total, ...rest}) => {
+		const durFmt = getDurFmt(locale);
+
 		return (
 			<div {...rest}>
+				<Media
+					controls
+					loop={loop}
+					mediaComponent={mediaComponent}
+					onUpdate={onUpdate}
+					ref={mediaRef}
+					source={source}
+				/>
 				<MediaSlider />
-				<Times />
-				<MediaControls />
+				<Times
+					current={currentTime}
+					formatter={durFmt}
+					total={total}
+				/>
+				<MediaControls
+					loop={loop}
+					onLoopChange={onLoopChange}
+					onPause={onPause}
+					onPlay={onPlay}
+					paused={paused}
+				/>
 			</div>
 		);
 	}
 });
 
-const MediaPlayer = Skinnable(MediaPlayerBase);
+/**
+ * Media player behaviors to apply to [MediaPlayerBase]{@link agate/MediaPlayer.MediaPlayerBase}.
+ *
+ * @class MediaPlayerExtended
+ * @memberof agate/MediaPlayer
+ * @hoc
+ * @private
+ */
+const MediaPlayerExtended = hoc((config, Wrapped) => { // eslint-disable-line no-unused-vars
```

Comment:
Maybe `MediaPlayerBehaviorDecorator`?
You could make this as hooks. But I don't mind.